dio/example/example.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry_dio/sentry_dio.dart';
+
+Future<void> main() async {
```

Comment:
can we make a pub project like https://github.com/getsentry/sentry-dart/tree/main/dart/example?
so we can compile and run easily on CI as well

Reply:
Sure. I've also included it in the Flutter example app, maybe that's already enough?

Reply:
mm I guess because of `paths-ignore`, we'd not have a failed CI when changes don't integrate well?
I still kinda prefer the pub project facility, but it's a personal opinion.

Reply:
we can do that in a different PR, so we don't block it.